### DISCUSSION TITLE
[功能请求] 请问是否可以增加一个 `展示页面信息` 组件

### DISCUSSION THREAD

**Discussion by Tinhone** [the1812/Bilibili-Evolved#4327] [功能建议 / Ideas]
_使用 分享 复制视频信息 复制页面信息 展示页面信息 作为关键字搜索，没有看到相关讨论。_

_原功能请求请查看编辑历史。_

## 目前期望
在功能面板上增加 `展示页面信息` 按钮，点击按钮后，使当前视频封面、链接、标题、详细信息被集中展示到一个弹窗中。

## 未来功能期望
- 兼容番剧、影视、直播等页面。
- 支持在组件设置中添加、修改、隐藏分享信息的各部分。
- 支持更多信息展示。

## 好处
视频数据一目了然，用户将能够十分方便地浏览、手动复制当前页面信息。

并且手动复制后粘贴到 QQ 等软件中的内容也会比单独复制链接更加详细，分享视频时利人利己。

### 例 1
![图片](https://github.com/the1812/Bilibili-Evolved/assets/81268244/53ec9b26-9152-4c7f-8be0-63acfc8e95ee)
https://www.bilibili.com/video/BV1DT411z7xK/
「永遠」× TV动画《死神 千年血战篇 诀别谭》第二季2023年7月开播
27.1万播放 · 总弹幕数355 · 2023-03-28 00:31:29

### 例 2
![图片](https://github.com/the1812/Bilibili-Evolved/assets/81268244/76a2d273-50a9-40ad-86a9-312ec6a6d9bb)
https://www.bilibili.com/video/BV16Z4y1H7NG/
相 机 大 战
652.6万播放 · 总弹幕数1.8万 · 2020-06-13 15:00:51 · 全站排行榜最高第4名 

## 其他
如果没有这个功能，我需要按以下步骤复制视频信息并分享：
1. 点开功能面板，点击查看封面，复制视频封面，粘贴到 QQ。
2. 点进地址栏，复制视频链接，粘贴到 QQ。
3. 复制视频标题与数据，粘贴到 QQ，并修改 B 站那丑陋的文字排版。
4. 确认没有改错后，发送。

如此分享视频费时费力，而我已经像这样重复操作过上百次了，因此来提个功能请求。

**Comment by the1812** [comment]
图片得咋复制? Markdown?

**Reply by Tinhone** [reply]
Markdown 是一种美妙的解决方案，可以为用户提供一个设置选项，这对博客类型的网站很有用

关于原生的图片复制或许可以参考下面的链接，不过看起来它们做不到同时复制图片和文字：
https://juejin.cn/post/6909379124679311368
https://blog.csdn.net/TooManyDdficulty/article/details/120140406

然后我搜了一圈发现，现在的功能请求从代码层面上看有些异想天开，我稍后修改一下期望

**Reply by the1812** [reply]
直接复制图片的话 我记得 Firefox 不支持

**Reply by Tinhone** [reply]
[Clipboard.write()](https://developer.mozilla.org/zh-CN/docs/Web/API/Clipboard/write) 确实不支持 Firefox，因此个人感觉可以退而求其次，组件负责集中显示信息，用户负责复制，也仍比原来便捷

而未来或许可以再制作一个插件，以方便 Chrome&Edge 用户进行快捷复制
